### ISSUE TITLE
Duplicate method/properties/events when showing 'inherited'

### ISSUE THREAD

**Issue by ascot21** [ember-learn/ember-api-docs#276]
When you check the 'inherited' (or 'protected') checkbox when viewing the API docs for a module, you see duplicated methods, properties and events.  See screenshot below:

![screen shot 2017-07-11 at 10 34 10 am](https://user-images.githubusercontent.com/578560/28081645-c8c04ff4-6624-11e7-9ddd-63e2920f41c6.png)

**Comment by toddjordan** [comment]
Likely showing documentation up the inheritance change.  We should show the local documentation if both exist.

**Comment by jenweber** [comment]
@toddjordan is file path depth a reasonable proxy for which inheritance is more local? Is there a better option besides filtering things based on that?

**Comment by toddjordan** [comment]
@jenweber this will probably need some work in https://github.com/ember-learn/ember-jsonapi-docs to provide enough information for the client to determine which item to use (or more likely to filter it out).  We need to look at what's available there.  Most likely we have enough info (since we did in the old ruby-based api site I think)

**Comment by jenweber** [comment]
Agreed. I spent some time last night inspecting the network requests to see what I could use, and came to the conclusion that I needed something external, whether an inheritance tree or different query results/structure.

I'll check it out.

**Comment by jenweber** [comment]
The last commit with the api docs in the website repo was `dfb2591970183e15b7d16dc12ebf1573c885917d`

Digging around there and attempting to read Ruby, it seems like methods/properties had a flag added of overwritten_from or extended_from that informed display filtering, and it was based on a consequential ordering of the YUIdoc info.

In cases of duplicates, I'm going to try preferring the 2nd item and see if that does the trick.

api_docs.rb
```ruby
    def classitems
      unless @classitems
        @classitems = {}

        # Go from root up
        parents = ([extends] + uses.reverse).compact
        parents.each do |parent|
          parent.classitems.each do |type, items|
            items.each do |_, item|
              item = item.clone
              @classitems[item['itemtype']] ||= {}
              existing = @classitems[item['itemtype']][item['name']]
              if existing
                next if existing['class'] == item['class']
                item['overwritten_from'] = existing
              else
                item['extended_from'] = parent
              end
              @classitems[item['itemtype']][item['name']] = item
            end
          end
        end

        self.class.data.classitems.each do |item|
          next if item['class'] != self.name

          item = item.clone
          @classitems[item['itemtype']] ||= {}
          if existing = @classitems[item['itemtype']][item['name']]
            item['overwritten_from'] = existing
          end
          @classitems[item['itemtype']][item['name']] = item
        end
      end

      @classitems
    end
```

Template
```
  <div class="meta">
      <% if item['overwritten_from'] %>
        Inherited from
        <%= api_class_link item['overwritten_from']['class'], "method_#{item['overwritten_from']['name']}" %>
        <% if item['file'] %>
          but overwritten in
        <% end %>
      <% else %>
        <% if item['extended_from'] %>
          Inherited from
          <%= api_class_link item['extended_from'], "method_#{item['name']}" %>
        <% else %>
          <% if item['file'] %>
            Defined in
          <% end %>
        <% end %>
      <% end %>
      <%= api_file_link item %>

    <% if item['since'] %>
      <br/>Available since <%= item['since'] %>
    <% end %>
  </div>
```

**Comment by sivakumar-kailasam** [comment]
Fixed via #301
